samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -35,56 +33,54 @@
+public class LocalityManager {
   private static final Logger log = LoggerFactory.getLogger(LocalityManager.class);
   private Map<String, Map<String, String>> containerToHostMapping = new HashMap<>();
   private final TaskAssignmentManager taskAssignmentManager;
   private final boolean writeOnly;
+  private CoordinatorStreamManager coordinatorStreamManager;
+  private static final String CONTAINER_PREFIX = "SamzaContainer-";
+
 
   /**
-   * Default constructor that creates a read-write manager
-   *
-   * @param coordinatorStreamProducer producer to the coordinator stream
-   * @param coordinatorStreamConsumer consumer for the coordinator stream
+   * Constructor that creates a read-write or write-only locality manager.
+   * 
+   * @param coordinatorStreamManager Coordinator stream manager.
+   * @param writeOnly True if write-only, otherwise false.
    */
-  public LocalityManager(CoordinatorStreamSystemProducer coordinatorStreamProducer,
-                         CoordinatorStreamSystemConsumer coordinatorStreamConsumer) {
-    super(coordinatorStreamProducer, coordinatorStreamConsumer, "SamzaContainer-");
-    this.writeOnly = coordinatorStreamConsumer == null;
-    this.taskAssignmentManager = new TaskAssignmentManager(coordinatorStreamProducer, coordinatorStreamConsumer);
+  public LocalityManager(CoordinatorStreamManager coordinatorStreamManager, boolean writeOnly) {
+    this.coordinatorStreamManager = coordinatorStreamManager;
+    this.writeOnly = writeOnly;
+    this.taskAssignmentManager = new TaskAssignmentManager(coordinatorStreamManager);
   }
 
-
   /**
-   * Special constructor that creates a write-only {@link LocalityManager} that only writes
-   * to coordinator stream in {@link SamzaContainer}
+   * Registers the locality manager with a source suffix that is container id
    *
-   * @param coordinatorStreamSystemProducer producer to the coordinator stream
+   * @param sourceSuffix the source suffix which is a container id
    */
-  public LocalityManager(CoordinatorStreamSystemProducer coordinatorStreamSystemProducer) {
-    this(coordinatorStreamSystemProducer, null);
+  public void register(String sourceSuffix) {
+    if (!writeOnly) {
+      coordinatorStreamManager.registerCoordinatorStreamConsumer();
+    }
+    coordinatorStreamManager.registerCoordinatorStreamProducer(CONTAINER_PREFIX + sourceSuffix);
   }
 
   /**
-   * This method is not supported in {@link LocalityManager}. Use {@link LocalityManager#register(String)} instead.
-   *
-   * @throws UnsupportedOperationException in the case if a {@link TaskName} is passed
+   * Ensures the coordinator stream is started.
    */
-  @Override
-  public void register(TaskName taskName) {
-    throw new UnsupportedOperationException("TaskName cannot be registered with LocalityManager");
+  public void start() {
+    if (!writeOnly) {
+      this.coordinatorStreamManager.startCoordinatorStreamConsumer();
```

Comment:
Other classes should not be able to manage lifecycle for CoordinatorStreamManager components.
Can we make the producer and consumer and their register/start/stop methods private to the CoordinatorStreamManager? 

Also, The coordinatorStreamManager should be started/stopped by the class that created it, not the classes that use it. Is that possible?